src/main/java/com/smiley/smileybackend/service/UserMedicalInfoService.java

```diff
@@ -0,0 +1,51 @@
+package com.smiley.smileybackend.service;
+
+import com.smiley.smileybackend.domain.Hospital;
+import com.smiley.smileybackend.domain.User;
+import com.smiley.smileybackend.domain.UserMedicalInfo;
+import com.smiley.smileybackend.dto.response.UserMedicalInfoSave;
+import com.smiley.smileybackend.dto.user.MedicalInfo;
+import com.smiley.smileybackend.repository.HospitalRepository;
+import com.smiley.smileybackend.repository.UserMedicalInfoRepository;
+import com.smiley.smileybackend.repository.UserRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class UserMedicalInfoService {
+    @Autowired
+    private UserMedicalInfoRepository userMedicalInfoRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private HospitalRepository hospitalRepository;
+
```

Comment:
@Autowired 대신 생성자 주입 방식으로 변경하는게 좋을 것 같습니다 !

```suggestion
public class UserMedicalInfoService {
     private UserMedicalInfoRepository userMedicalInfoRepository,
     private UserRepository userRepository,
     private HospitalRepository hospitalRepository

     public UserMedicalInfoService(
          UserMedicalInfoRepository userMedicalInfoRepository,
          UserRepository userRepository,
          HospitalRepository hospitalRepository
    ){
         this.userMedicalInfoRepository = userMedicalInfoRepository;
         this.userRepository = userRepository;
         this.hospitalRepository = hospitalRepository
    }         
}



```

Reply:
위의 의견과 동일합니다. @Autowired 어노테이션을 사용하면 순환 참조가 발생하는 경우가 있을 수 있습니다. 하지만 애플리케이션은 구동되기 때문에 알아차리지 못할 수 있지만 생성자 주입을 사용하면 발생할 수 있는 오류를 사전에 알 수 있을 뿐만 아니라 다양한 장점이 있기 때문에 생성자 주입 방식으로 바꾸는 것이 좋을 것 같습니다!

Reply:
생성자 주입을 @Autowired보다 권장하는 이유에 대한 참고 링크입니다.
https://madplay.github.io/post/why-constructor-injection-is-better-than-field-injection